### DISCUSSION TITLE
tpch_benchmark failed with segment fault

### DISCUSSION THREAD

**Discussion by fandebiao01** [facebookincubator/velox#11793] [Q&A]
I followed the README to build velox in ubuntun22.04 container, but the tpc_benchmark failed immediately.
Thanks for your help in advance!

```
 ./_build/release/velox/benchmarks/tpch/velox_tpch_benchmark  -data_path=.
terminate called after throwing an instance of 'std::out_of_range'
  what():  _Map_base::at
*** Aborted at 1733726531 (Unix time, try 'date -d @1733726531') ***
*** Signal 6 (SIGABRT) (0x1ce73) received by PID 118387 (pthread TID 0x7faae94343c0) (linux TID 118387) (maybe from PID 118387, UID 0) (code: -6), stack trace: ***
    @ 0000000006fd2191 _ZN5folly10symbolizer12_GLOBAL__N_113signalHandlerEiP9siginfo_tPv
                       velox/deps-download/folly/folly/experimental/symbolizer/SignalHandler.cpp:453
    @ 000000000004251f (unknown)
    @ 00000000000969fc pthread_kill
    @ 0000000000042475 raise
    @ 00000000000287f2 abort
    @ 00000000000a2b9d (unknown)
    @ 00000000000ae20b (unknown)
    @ 00000000000ae276 _ZSt9terminatev
    @ 00000000000ae4d7 __cxa_throw
    @ 0000000006f8702a __cxa_throw
                       velox/deps-download/folly/folly/debugging/exception_tracer/ExceptionTracerLib.cpp:159
    @ 00000000000a549f _ZSt20__throw_out_of_rangePKc
    @ 00000000026de3f5 _ZNKSt8__detail9_Map_baseINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEESt4pairIKS6_N8facebook5velox4exec4test17TpchTableMetadataEESaISE_ENS_10_Select1stESt8equal_toIS6_ESt4hashIS6_ENS_18_Mod_range_hashingENS_20_Default_ranged_hashENS_20_Prime_rehash_policyENS_17_Hashtable_traitsILb1ELb0ELb1EEELb1EE2atERS8_
    @ 00000000026e0057 _ZNK8facebook5velox4exec4test16TpchQueryBuilder10getRowTypeERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEERKSt6vectorIS9_SaIS9_EE
    @ 00000000026a2ef0 _ZNK8facebook5velox4exec4test16TpchQueryBuilder9getQ1PlanEv
    @ 00000000026d8af4 _ZNK8facebook5velox4exec4test16TpchQueryBuilder12getQueryPlanEi
    @ 00000000025b008e _ZNSt17_Function_handlerIFN5folly6detail12TimeIterDataEjEZNS1_17BenchmarkingStateINSt6chrono3_V212system_clockEE12addBenchmarkIZNS8_12addBenchmarkIRNL22follyBenchmarkUnused16MUlvE_EEENSt9enable_ifIX14is_invocable_vIT_EEvE4typeEPKcNS0_5RangeISI_EEOSE_EUljE_EENSD_IX14is_invocable_vISE_jEEvE4typeESI_SK_SL_EUljE_E9_M_invokeERKSt9_Any_dataOj
    @ 0000000006f34c86 _ZN5follyL29runBenchmarkGetNSPerIterationERKSt8functionIFNS_6detail12TimeIterDataEjEEd
                       /usr/include/c++/11/bits/std_function.h:590
                       -> velox/deps-download/folly/folly/Benchmark.cpp
    @ 0000000006f3bdd1 _ZN5folly12_GLOBAL__N_128runBenchmarksWithPrinterImplEPNS0_23BenchmarkResultsPrinterERKNS0_15BenchmarksToRunE
                       velox/deps-download/folly/folly/Benchmark.cpp:773
    @ 0000000006f3d968 _ZNK5folly6detail21BenchmarkingStateBase24runBenchmarksWithPrinterINS_12_GLOBAL__N_123BenchmarkResultsPrinterEEESt4pairISt3setINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEESt4lessISC_ESaISC_EESt6vectorINS0_15BenchmarkResultESaISI_EEEPT_
                       velox/deps-download/folly/folly/Benchmark.cpp:880
    @ 0000000006f3e94f _ZN5folly13runBenchmarksEv
                       velox/deps-download/folly/folly/Benchmark.cpp:920
    @ 00000000025b508d _Z17tpchBenchmarkMainv
    @ 00000000025874cd main
    @ 0000000000029d8f (unknown)
    @ 0000000000029e3f __libc_start_main
    @ 00000000025afa74 _start
Aborted (core dumped)
```

**Comment by karthikeyann** [comment]
This PR https://github.com/facebookincubator/velox/pull/11833 will fix the TPCH benchmark crash.
Also, ensure there is tpch data in parquet format (Default) in data_path directory.

**Reply by fandebiao01** [reply]
Thank you for the quick fix. I checkouted your branch and  followed https://github.com/ljishen/tpch-data to generate tpch-data in parquet format, but the test is still failling..

```
ls data
customer.parquet  lineitem.parquet  nation.parquet  orders.parquet  part.parquet  partsupp.parquet  region.parquet  supplier.parquet
root@74938db9374c:/data00/xqj/spark/velox_new/velox/_build/release/velox/benchmarks/tpch# ./velox_tpch_benchmark  -data_path=data
terminate called after throwing an instance of 'std::out_of_range'
  what():  _Map_base::at
*** Aborted at 1734602449 (Unix time, try 'date -d @1734602449') ***
*** Signal 6 (SIGABRT) (0x703a) received by PID 28730 (pthread TID 0x7f47750b43c0) (linux TID 28730) (maybe from PID 28730, UID 0) (code: -6), stack trace: ***
    @ 0000000006ff5001 _ZN5folly10symbolizer12_GLOBAL__N_113signalHandlerEiP9siginfo_tPv
                       /data00/xqj/spark/velox/deps-download/folly/folly/experimental/symbolizer/SignalHandler.cpp:453
    @ 000000000004251f (unknown)
    @ 00000000000969fc pthread_kill
    @ 0000000000042475 raise
    @ 00000000000287f2 abort
    @ 00000000000a2b9d (unknown)
    @ 00000000000ae20b (unknown)
    @ 00000000000ae276 _ZSt9terminatev
    @ 00000000000ae4d7 __cxa_throw
    @ 0000000006fa9e9a __cxa_throw
                       /data00/xqj/spark/velox/deps-download/folly/folly/debugging/exception_tracer/ExceptionTracerLib.cpp:159
    @ 00000000000a549f _ZSt20__throw_out_of_rangePKc
    @ 00000000026ef095 _ZNKSt8__detail9_Map_baseINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEESt4pairIKS6_N8facebook5velox4exec4test17TpchTableMetadataEESaISE_ENS_10_Select1stESt8equal_toIS6_ESt4hashIS6_ENS_18_Mod_range_hashingENS_20_Default_ranged_hashENS_20_Prime_rehash_policyENS_17_Hashtable_traitsILb1ELb0ELb1EEELb1EE2atERS8_
    @ 00000000026f0cf7 _ZNK8facebook5velox4exec4test16TpchQueryBuilder10getRowTypeERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEERKSt6vectorIS9_SaIS9_EE
    @ 00000000026b3b90 _ZNK8facebook5velox4exec4test16TpchQueryBuilder9getQ1PlanEv
    @ 00000000026e9794 _ZNK8facebook5velox4exec4test16TpchQueryBuilder12getQueryPlanEi
    @ 00000000025c0a6e _ZNSt17_Function_handlerIFN5folly6detail12TimeIterDataEjEZNS1_17BenchmarkingStateINSt6chrono3_V212system_clockEE12addBenchmarkIZNS8_12addBenchmarkIRNL22follyBenchmarkUnused16MUlvE_EEENSt9enable_ifIX14is_invocable_vIT_EEvE4typeEPKcNS0_5RangeISI_EEOSE_EUljE_EENSD_IX14is_invocable_vISE_jEEvE4typeESI_SK_SL_EUljE_E9_M_invokeERKSt9_Any_dataOj
    @ 0000000006f57af6 _ZN5follyL29runBenchmarkGetNSPerIterationERKSt8functionIFNS_6detail12TimeIterDataEjEEd
                       /usr/include/c++/11/bits/std_function.h:590
                       -> /data00/xqj/spark/velox/deps-download/folly/folly/Benchmark.cpp
    @ 0000000006f5ec41 _ZN5folly12_GLOBAL__N_128runBenchmarksWithPrinterImplEPNS0_23BenchmarkResultsPrinterERKNS0_15BenchmarksToRunE
                       /data00/xqj/spark/velox/deps-download/folly/folly/Benchmark.cpp:773
    @ 0000000006f607d8 _ZNK5folly6detail21BenchmarkingStateBase24runBenchmarksWithPrinterINS_12_GLOBAL__N_123BenchmarkResultsPrinterEEESt4pairISt3setINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEESt4lessISC_ESaISC_EESt6vectorINS0_15BenchmarkResultESaISI_EEEPT_
                       /data00/xqj/spark/velox/deps-download/folly/folly/Benchmark.cpp:880
    @ 0000000006f617bf _ZN5folly13runBenchmarksEv
                       /data00/xqj/spark/velox/deps-download/folly/folly/Benchmark.cpp:920
    @ 00000000025c5a6d _Z17tpchBenchmarkMainv
    @ 0000000002597ead main
    @ 0000000000029d8f (unknown)
    @ 0000000000029e3f __libc_start_main
    @ 00000000025c0454 _start
Aborted (core dumped)
root@74938db9374c:/data00/xqj/spark/velox_new/velox/_build/release/velox/benchmarks/tpch# git branch
* fix-tpch_reader_crash
  main
```

**Reply by karthikeyann** [reply]
Is this issue still present?

**Reply by TheCandle** [reply]
> Thank you for the quick fix. I checkouted your branch and followed https://github.com/ljishen/tpch-data to generate tpch-data in parquet format, but the test is still failling..
> 
> ```
> ls data
> customer.parquet  lineitem.parquet  nation.parquet  orders.parquet  part.parquet  partsupp.parquet  region.parquet  supplier.parquet
> root@74938db9374c:/data00/xqj/spark/velox_new/velox/_build/release/velox/benchmarks/tpch# ./velox_tpch_benchmark  -data_path=data
> terminate called after throwing an instance of 'std::out_of_range'
>   what():  _Map_base::at
> *** Aborted at 1734602449 (Unix time, try 'date -d @1734602449') ***
> *** Signal 6 (SIGABRT) (0x703a) received by PID 28730 (pthread TID 0x7f47750b43c0) (linux TID 28730) (maybe from PID 28730, UID 0) (code: -6), stack trace: ***
>     @ 0000000006ff5001 _ZN5folly10symbolizer12_GLOBAL__N_113signalHandlerEiP9siginfo_tPv
>                        /data00/xqj/spark/velox/deps-download/folly/folly/experimental/symbolizer/SignalHandler.cpp:453
>     @ 000000000004251f (unknown)
>     @ 00000000000969fc pthread_kill
>     @ 0000000000042475 raise
>     @ 00000000000287f2 abort
>     @ 00000000000a2b9d (unknown)
>     @ 00000000000ae20b (unknown)
>     @ 00000000000ae276 _ZSt9terminatev
>     @ 00000000000ae4d7 __cxa_throw
>     @ 0000000006fa9e9a __cxa_throw
>                        /data00/xqj/spark/velox/deps-download/folly/folly/debugging/exception_tracer/ExceptionTracerLib.cpp:159
>     @ 00000000000a549f _ZSt20__throw_out_of_rangePKc
>     @ 00000000026ef095 _ZNKSt8__detail9_Map_baseINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEESt4pairIKS6_N8facebook5velox4exec4test17TpchTableMetadataEESaISE_ENS_10_Select1stESt8equal_toIS6_ESt4hashIS6_ENS_18_Mod_range_hashingENS_20_Default_ranged_hashENS_20_Prime_rehash_policyENS_17_Hashtable_traitsILb1ELb0ELb1EEELb1EE2atERS8_
>     @ 00000000026f0cf7 _ZNK8facebook5velox4exec4test16TpchQueryBuilder10getRowTypeERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEERKSt6vectorIS9_SaIS9_EE
>     @ 00000000026b3b90 _ZNK8facebook5velox4exec4test16TpchQueryBuilder9getQ1PlanEv
>     @ 00000000026e9794 _ZNK8facebook5velox4exec4test16TpchQueryBuilder12getQueryPlanEi
>     @ 00000000025c0a6e _ZNSt17_Function_handlerIFN5folly6detail12TimeIterDataEjEZNS1_17BenchmarkingStateINSt6chrono3_V212system_clockEE12addBenchmarkIZNS8_12addBenchmarkIRNL22follyBenchmarkUnused16MUlvE_EEENSt9enable_ifIX14is_invocable_vIT_EEvE4typeEPKcNS0_5RangeISI_EEOSE_EUljE_EENSD_IX14is_invocable_vISE_jEEvE4typeESI_SK_SL_EUljE_E9_M_invokeERKSt9_Any_dataOj
>     @ 0000000006f57af6 _ZN5follyL29runBenchmarkGetNSPerIterationERKSt8functionIFNS_6detail12TimeIterDataEjEEd
>                        /usr/include/c++/11/bits/std_function.h:590
>                        -> /data00/xqj/spark/velox/deps-download/folly/folly/Benchmark.cpp
>     @ 0000000006f5ec41 _ZN5folly12_GLOBAL__N_128runBenchmarksWithPrinterImplEPNS0_23BenchmarkResultsPrinterERKNS0_15BenchmarksToRunE
>                        /data00/xqj/spark/velox/deps-download/folly/folly/Benchmark.cpp:773
>     @ 0000000006f607d8 _ZNK5folly6detail21BenchmarkingStateBase24runBenchmarksWithPrinterINS_12_GLOBAL__N_123BenchmarkResultsPrinterEEESt4pairISt3setINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEESt4lessISC_ESaISC_EESt6vectorINS0_15BenchmarkResultESaISI_EEEPT_
>                        /data00/xqj/spark/velox/deps-download/folly/folly/Benchmark.cpp:880
>     @ 0000000006f617bf _ZN5folly13runBenchmarksEv
>                        /data00/xqj/spark/velox/deps-download/folly/folly/Benchmark.cpp:920
>     @ 00000000025c5a6d _Z17tpchBenchmarkMainv
>     @ 0000000002597ead main
>     @ 0000000000029d8f (unknown)
>     @ 0000000000029e3f __libc_start_main
>     @ 00000000025c0454 _start
> Aborted (core dumped)
> root@74938db9374c:/data00/xqj/spark/velox_new/velox/_build/release/velox/benchmarks/tpch# git branch
> * fix-tpch_reader_crash
>   main
> ```

Do you solve it?
